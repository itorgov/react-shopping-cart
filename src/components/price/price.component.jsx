export const Price = ({value}) => {
    const formatter = new Intl.NumberFormat('en-US', {style: 'currency', currency: 'USD'});

    return formatter.format(value / 100);
};
