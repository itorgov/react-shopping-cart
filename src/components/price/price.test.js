import React from 'react';
import {shallow} from "enzyme";

import {Price} from './price.component';

describe('<Price />', () => {
    it('expect to render', () => {
        expect(shallow(<Price />)).toMatchSnapshot();
    });

    it('has to divide value by 100 and format it to US dollars', () => {
        expect(shallow(<Price value={2233} />).text()).toEqual('$22.33');
    })
});
