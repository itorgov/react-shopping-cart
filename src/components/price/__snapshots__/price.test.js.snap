// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> expect to render 1`] = `"$NaN"`;
